service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/ControlledAzureDiskHandler.java

```diff
@@ -17,6 +19,17 @@ public static ControlledAzureDiskHandler getHandler() {
   /** {@inheritDoc} */
   @Override
   public WsmResource makeResourceFromDb(DbResource dbResource) {
-    return new ControlledAzureDiskResource(dbResource);
+    ControlledAzureDiskAttributes attributes =
+        DbSerDes.fromJson(dbResource.getAttributes(), ControlledAzureDiskAttributes.class);
+
+    var resource =
+        ControlledAzureDiskResource.builder()
+            .diskName(attributes.getDiskName())
+            .region(attributes.getRegion())
+            .size(attributes.getSize())
+            .common(new ControlledResourceFields(dbResource))
+            .build();
+    resource.validate();
```

Comment:
The `builder.build()` method already calls `validate()`, you don't need a second call here. Same for other resource types

Reply:
Good catch!